### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by leotrs**
Some progress in documentation. The changes in this PR can be summarized in three types:

1. The API reference pages are autogenerated, and the tables looked a bit tight/ugly. So this PR adds 10px padding on them by using a custom css file. This accounts for the diff in `custom.css` and `conf.py`.

2. The autogenerated rst files themselves were littered with blank lines and were hard to read (at least for me). This PR changes the template minimally in order to fix that. This accounts for the diff in `class.rst`.

3. I wrote docstrings for Mobject, Group, Mobject.add and Mobject.remove. This accounts for the diff in `mobject.py`.

4. I added module-level docstrings to start populating the API reference page. This accounts for the diff in all remaining files.

EDIT: apologies for the messy commit history!

### CODE REVIEW

**Comment by leotrs** [comment]
Restyled failed during forking/branching for some reason, and CodeFactor is just being silly. This PR is ready for review.

File: manim/animation/animation.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+"""Animate mobjects."""
```
Comment (kolibril13) [inline manim/animation/animation.py:1]: Animate Mobjects (capitalized)

File: manim/animation/fading.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+"""Fading in and out of view."""
```
Comment (kolibril13) [inline manim/animation/fading.py:1]: maybe "of view"   to "on screen-view"?

File: manim/mobject/mobject.py [hunk 3]
```diff
@@ -30,8 +28,13 @@
 
 
 class Mobject(Container):
-    """
-    Mathematical Object
+    """Mathematical Object: base class for all objects shown on screen.
```
Comment (kolibril13) [inline manim/mobject/mobject.py:31]: base class for all objects that can be constructed and can be added on screen.

**Comment by kolibril13** [comment]
Do we have yet conventions for names of Mobjects that are
* a) constructed but not added to a scene
* b) constructed and added to a scene? 

Because at the moment when 
talking about b)
we have : "Fading in and out of view" , "arrival on screen" , "The mobjects are added to self.submobjects" 
when talking about a)
"Mathematical Object: base class for all objects shown on screen."

Maybe it would be nice to have here something like "constructed Mobjects" and " displayed Mobjects"

**Comment by leotrs** [comment]
> Do we have yet conventions for names of Mobjects that are
> 
> * a) constructed but not added to a scene
> * b) constructed and added to a scene?
> 
> Because at the moment when
> talking about b)
> we have : "Fading in and out of view" , "arrival on screen" , "The mobjects are added to self.submobjects"
> when talking about a)
> "Mathematical Object: base class for all objects shown on screen."
> 
> Maybe it would be nice to have here something like "constructed Mobjects" and " displayed Mobjects"

I'm _so_happy you bring this up. We should definitely have conventions so we are clear and consistent across the documentation. I like "displayed".

Relatedly, I'll just share another convention I was using when writing this PR: Use `Mobject` only when referring to the actual class. When referring to a single instance of Mobject, whether it is on screen or not, I tried using "mobject" or plural "mobjects".

With these two conventions, we could change the sentences you highlighted to 

+ For `fading.py`: "Display a mobject by fading it in, or stop displaying it by fading it out."
+ Change "arrival" to "the first time it is displayed"
+ "The mobjects are added to self.submobjects" can be changed to "The mobjects are added to self.submobjects. Note this does not display them on screen."
+ "Animate mobjects" would remain with "mobjects" not capitalized.


What do you think?

**Comment by kolibril13** [comment]
> With these two conventions, we could change the sentences you highlighted to
> 
> * For `fading.py`: "Display a mobject by fading it in, or stop displaying it by fading it out."
> * Change "arrival" to "the first time it is displayed"
> * "The mobjects are added to self.submobjects" can be changed to "The mobjects are added to self.submobjects. Note this does not display them on screen."
> * "Animate mobjects" would remain with "mobjects" not capitalized.
> 
> What do you think?

I like this!
maybe we can also an additional word for "The mobjects are added to self.submobjects", that makes clear that they are now part of the scene, but not displayed.

And do we want to make a new document for naming conventions?
Maybe we can put it in the same file as the changelog.

**Comment by leotrs** [comment]
> maybe we can also an additional word for "The mobjects are added to self.submobjects", that makes clear that they are now part of the scene, but not displayed.

I'm not sure I understood you correctly, so I wanted to check. Here's a scene:

```python
import manim
class TestScene(manim.Scene):
    def construct(self):
        c = manim.Circle()
        s = manim.Square()
        self.add(c)
        c.add(s)
        self.wait(1)

```
If you run this, it shows both the circle and the square. So the documentation should say something like
```
Add mobjects as submobjects.

The mobjects are added to self.submobjects.  If the parent mobject is displayed, the newly-added submobjects will also be displayed (i.e. they are automatically added to the parent Scene).
```


> And do we want to make a new document for naming conventions?
> Maybe we can put it in the same file as the changelog.

I think maybe the wiki is a better place for this. What do you think?

**Comment by kolibril13** [comment]
> Add mobjects as submobjects.
> 
> The mobjects are added to self.submobjects.  If the parent mobject is 
> displayed in a Scene, the newly-added submobjects will also be displayed 
> (i.e. they are automatically added to the parent Scene).

Yes, but now that I read it, I think it is better to keep it simple, without the extra sentence that might cause more confusion then benefit.

To make this sentence clear, maybe we can add your example later to the documentation examples

**Comment by leotrs** [comment]
> Yes, but now that I read it, I think it is better to keep it simple, without the extra sentence that might cause more confusion then benefit.

That's fair. I'll add that sentence in the notes section instead.

**Comment by leotrs** [comment]
@kolibril13 I just pushed a bunch of changes after your review. Let me know if you think this is ready to be merged!

File: manim/camera/camera.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+"""A Camera converts a Scene into visual data."""
```
Comment (kolibril13) [inline manim/camera/camera.py:1]: Is "data" the right word here?
Maybe "A Camera converts a Mobjects of a Scene into displayed animations" instead?

File: manim/mobject/mobject.py [hunk 5]
```diff
@@ -1097,6 +1150,8 @@ def set_z_index_by_z_coordinate(self):
 
 
 class Group(Mobject):
+    """Groups together multiple MObjects."""
```
Comment (kolibril13) [inline manim/mobject/mobject.py:1153]: small "o" -> Mobjects

File: manim/mobject/svg/svg_mobject.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+"""Mobjects generated from SVG files."""
```
Comment (kolibril13) [inline manim/mobject/svg/svg_mobject.py:1]: Better singular:
Mobject generated from a SVG file

File: manim/scene/scene.py [hunk 7]
```diff
@@ -1,3 +1,4 @@
+"""A Scene is the canvas of your animation."""
```
Comment (huguesdevimeux) [inline manim/scene/scene.py:1]: **Nitpick:** Shakespeare here. I would change 'your' to maybe "the"

**Review by huguesdevimeux — APPROVED** [review]
SGTM !

File: manim/animation/creation.py [hunk 8]
```diff
@@ -1,3 +1,4 @@
+"""Display a mobject by fading it in, or stop displaying it by fading it out."""
```
Comment (Aathish04) [inline manim/animation/creation.py:1]:
```suggestion
"""Animate the addition or removal of a mobject from a Scene."""
```
Nit: This sounds too specific, as if it only applies to `FadeIn` and `FadeOut`.

**Review by Aathish04 — CHANGES_REQUESTED** [review]

File: manim/camera/camera.py (hunk 4 above)
Comment (leotrs) [reply manim/camera/camera.py:1]: Hmm I don't like "displayed animations". We are using the verb "display" for adding a mobject or animation to the scene. What a Camera does is it takes all of that information and make a numpy array that contains RGB values per pixel. That's why I chose data. How about:

"A Camera converts the mobjects contained in a Scene into an array of pixels"

File: manim/animation/creation.py (hunk 8 above)
Comment (leotrs) [reply manim/animation/creation.py:1]: lol I didn't catch that thanks

Comment (leotrs) [reply manim/animation/creation.py:1]: I changed it to  `Animate the display or removal of a mobject from a scene`, as we are trying to use the verb "display" for adding a mobject to the scene

**Comment by leotrs** [comment]
Most comments have been addressed, just waiting on confirmation by @kolibril13 :)

**Review by Aathish04 — APPROVED** [review]

**Comment by kolibril13** [comment]
Approved!

File: manim/mobject/mobject.py [hunk 9]
```diff
@@ -69,6 +72,37 @@ def generate_points(self):
         pass
 
     def add(self, *mobjects):
+        """Add mobjects as submobjects.
+
+        The mobjects are added to self.submobjects.
+
+        Parameters
+        ----------
+        mobjects : List[:class:`Mobject`]
```
Comment (PgBiel) [inline manim/mobject/mobject.py:81]: we aren't expecting lists of mobjects, right?
```suggestion
        mobjects : :class:`Mobject`
```

File: manim/mobject/mobject.py [hunk 10]
```diff
@@ -69,6 +72,37 @@ def generate_points(self):
...
+        Notes
+        -----
+        A mobject cannot contain itself, and it cannot contain a submobject
+        more than once.  If the parent mobject is displayed, the newly-added
+        submobjects will also be displayed (i.e. they are automatically added
+        to the parent Scene).
+
+        See Also
+        --------
+        :meth:`~Mobject.remove`
```
Comment (PgBiel) [inline manim/mobject/mobject.py:103]: i mean this can be shortened to just "remove" but we can let it be i guess

File: manim/mobject/mobject.py [hunk 11]
```diff
@@ -80,6 +114,25 @@ def add_to_back(self, *mobjects):
         return self
 
     def remove(self, *mobjects):
+        """Remove submobjects.
+
+        The mobjects are removed from self.submobjects, if they exist.
+
+        Parameters
+        ----------
+        mobjects : List[:class:`Mobject`]
```
Comment (PgBiel) [inline manim/mobject/mobject.py:123]:
```suggestion
        mobjects : :class:`Mobject`
```

File: manim/mobject/mobject.py (hunk 9 above)
Comment (leotrs) [reply manim/mobject/mobject.py:81]: `mobjects` is always a list, no? How to specify it is packing all arguments passed?

**Pull request merged**